### ISSUE TITLE
Adding erlang:process_info/2 statistics

### ISSUE THREAD

**Issue by gord1anknot** [ferd/vmstats#9]
Hello,

I'm working on troubleshooting a production Erlang app. I'd like to be able to send statistics about specific registered processes, gathered using process_info/2, and send them through StatsD. Specifically, I'm interested in {message_queue_len, MessageQueueLen}, {reductions, Number}, and potentially several of the memory and gc related stats, depending on if we spot the issue right away or not.

So far vmstats has worked well for me, and I'm very happy to code it up, add tests, configuration, docs, etc, and submit a pr for you review, but only if you're interested in taking it, of course :)

let me know what you think! Thanks!

**Comment by ferd** [comment]
How many processes would you expect to track? What happens if they go away? What if they come back? And so on?

**Comment by gord1anknot** [comment]
Thanks for the super quick reply!

For my immediate use case, I can assume that the number of processes and the atoms of their registered names are known in advance, before starting the beam vm, which allows them to be passed as otp application parameters or something like that.

I've scoped out what I need in a test environment with Observer, and now I pretty much want a safer subset of that Observer data, to use in prod if possible. I've already created a little POC that makes some assumptions as follows:

- there's about 20-40 processes I'm interested in, and I know their registered names
- somehow a list of atoms is provided to vmstats

In the POC I've made, I go through that list of atoms, calling process_info/2 on each. I have another erlang term (known at VM start) with the stats I want to gather, and pass that list as the second argument to process_info/2. If the output of that is 'undefined', I do nothing, otherwise, I create a list with some sub-set of stats on some sub-set of those registered names. Basically, it's a monad. I pass the non-undefined pid from that process as gauges to statsd, meaning that the lack of information on a particular collection run, if a process disappeared, doesn't mean anything from StatsD's perspective, leaving a zero.

There's a couple of problems with that:

- people other than me may not know what processes they care about, so a more general implementation I contribute here should either handle dynamically discovering processes or explicitly not handle it, leaving the list-of-atoms-that-are-registered-names approach
- if the processes you're tracking include statsderl's ports than there's a potential for feedback I'd expect, depending on the stat collected, so my implementation could filter those out
- calling process_info/2 20-40 times a second is not as good of an idea as I think it might be ;)
